var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -193,39 +212,54 @@ class PyTensorflow(Package, CudaPackage):
     depends_on('py-protobuf@3.0.0', type=('build', 'run'), when='@0.11.0')
     depends_on('py-protobuf@3.0.0b2', type=('build', 'run'), when='@0.7.1:0.10')
     depends_on('py-protobuf@3.0.0a3', type=('build', 'run'), when='@0.6:0.7.0')
-    depends_on('protobuf')
-    depends_on('flatbuffers+python@1.12.0:1.12', type=('build', 'run'), when='@2.4.0:')
+    depends_on('protobuf@:3.12', when='@:2.4')
+    depends_on('protobuf@:3.17')
+    depends_on('flatbuffers+python@1.12:2.9', type=('build', 'run'), when='@2.7:')
+    depends_on('flatbuffers+python@1.12.0:1.12', type=('build', 'run'), when='@2.4.0:2.6')
+
     # tensorboard
-    # tensorflow-estimator
-    depends_on('py-termcolor@1.1.0:1.1', type=('build', 'run'), when='@2.4.0:')
-    depends_on('py-termcolor@1.1.0:', type=('build', 'run'), when='@1.6:2.3')
-    depends_on('py-wrapt@1.12.1:1.12', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-tensorboard@2.5.0:2.5', type=('build', 'run'), when='@2.5')
+    depends_on('py-tensorboard@2.6.0:2.6', type=('build', 'run'), when='@2.6')
+    depends_on('py-tensorboard@2.7.0:2.7', type=('build', 'run'), when='@2.7')
```

Comment:
You've added a tensorboard dep that we've previously removed due to a circular dependency, but we're still commenting out the tensorboard dep in `setup.py` in `post_configure_fixes`, should we add a version check to guard that? Also, the `setup.py` also lists `keras` and `tensorflow-io-gcs-filesystem` as deps. If those have circular deps, you can try commenting them out. But if we don't install them, setuptools/pip will, which we don't want.

Reply:
> `setup.py` says `~= 2.6` which translates to `2.6:2` in Spack, I'm guessing the other versions are wrong too

I understand that, but my understanding is that the versions need to match to the minor version. The `@2.6:2` would allow tensorboard-2.7 to be used with tensorflow-2.6, which I believe will cause issues.

Reply:
Gotcha, might want to submit a PR to the TF devs to use `~= 2.6.0` or `== 2.6.*` instead.

Reply:
I just committed a change to adjust the dependencies listed in the setup.py file for 

- tensorboard
- tensorflow-estimator
- keras
- tensorflow-io-gcs-filesystem

The last one is new for tensorflow-2.7 and I think would best be added in a separate PR.